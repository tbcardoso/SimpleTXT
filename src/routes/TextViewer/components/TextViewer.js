import React from 'react';
import {
	StyleSheet,
	View,
} from 'react-native';
import TxtView from '../../../components/TxtView/TxtView';
import Icon from 'react-native-vector-icons/MaterialIcons';


const sampleText = `
  ____  _                 _     _______  _______ 
 / ___|(_)_ __ ___  _ __ | | __|_   _\\ \\/ /_   _|
 \\___ \\| | '_ ' _ \\| '_ \\| |/ _ \\| |  \\  /  | |  
  ___) | | | | | | | |_) | |  __/| |  /  \\  | |  
 |____/|_|_| |_| |_| .__/|_|\\___||_| /_/\\_\\ |_|  
                   |_|                           
`;


export default class TextViewer extends React.Component {

	_onActionSelected = (position) => {
		switch (position) {
			case 0:
				this.props.onOpenFile();
				break;
		}
	}

	render() {
		const pos = this.props.filePath.lastIndexOf('/');
		const fileName = this.props.filePath.substring(Math.max(0, pos+1));
		const title = fileName || 'SimpleTXT';
		const fileContent = fileName ? this.props.fileContent : sampleText;
		return (
			<View style={{ flex: 1 }}>
				<Icon.ToolbarAndroid
					title={title}
					style={styles.toolbar}
					actions={toolbarActions}
					onActionSelected={this._onActionSelected} />
				<TxtView style={{ flex: 1 }}
					filePath={this.props.filePath} fileContent={fileContent} />
			</View>
		);
	}
}

TextViewer.propTypes = {
  filePath: React.PropTypes.string.isRequired,
	fileContent: React.PropTypes.string,
	onOpenFile: React.PropTypes.func.isRequired,
};


const toolbarActions = [
	{ title: 'Open File', iconName: 'insert-drive-file', iconSize: 30, show: 'always' },
];


const styles = StyleSheet.create({
	toolbar: {
		backgroundColor: '#a9a9a9',
		height: 56,
	},
});