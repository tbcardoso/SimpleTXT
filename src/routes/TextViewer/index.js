import TextViewer from './components/TextViewer'
import reducer from './modules/reducer'

export default {
	title: 'TextViewer',
	component: TextViewer,
	reducer: reducer
}
