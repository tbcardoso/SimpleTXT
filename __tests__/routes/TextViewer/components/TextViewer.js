/* eslint-env jest */
import React from 'react';
import 'react-native';
import TextViewer from '../../../../src/routes/TextViewer/components/TextViewer';
import renderer from 'react-test-renderer';

jest.mock('react-native-vector-icons/MaterialIcons', () => ({ ToolbarAndroid: 'ToolbarAndroid' }));

describe('Component TextViewer', () => {
	test('renders correctly', () => {
		const tree = renderer.create(
			<TextViewer filePath=''
				onOpenFile={() => { }}
				onAbout={() => { }} />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
