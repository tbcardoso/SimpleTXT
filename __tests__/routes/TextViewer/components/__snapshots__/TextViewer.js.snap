// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TextViewer renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ToolbarAndroid
    actions={
      Array [
        Object {
          "iconName": "insert-drive-file",
          "iconSize": 30,
          "show": "always",
          "title": "Open File",
        },
        Object {
          "show": "never",
          "title": "About",
        },
      ]
    }
    onActionSelected={[Function]}
    style={
      Object {
        "backgroundColor": "#607d8b",
        "height": 56,
      }
    }
    title="SimpleTXT"
    titleColor="#ffffff"
  />
  <STTxtView
    fileContent=""
    filePath=""
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 3,
      }
    }
  />
</View>
`;
